src/main/java/receive_a_message/SmsHelloWorld.java

```
@@ -13,19 +13,65 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+
+class IncomingSmsRequest
+{
+    @JsonProperty("accountId")
+    public String accountId;
+    @JsonProperty("applicationId")
+    public String applicationId;
+    @JsonProperty("direction")
+    public String direction;
+    @JsonProperty("from")
+    public String from;
+    @JsonProperty("messageId")
+    public String messageId;
+    @JsonProperty("phoneNumberId")
+    public String phoneNumberId;
+    @JsonProperty("requestType")
+    public String requestType;
+    @JsonProperty("status")
+    public String status;
+    @JsonProperty("text")
+    public String text;
+    @JsonProperty("to")
+    public String to;
+
+    public IncomingSmsRequest() {
+    }
+
+    public IncomingSmsRequest(String accountId, String applicationId, String direction, String from, String messageId, String phoneNumberId, String requestType, String status, String text, String to) {
+        this.accountId = accountId;
+        this.applicationId = applicationId;
+        this.direction = direction;
+        this.from = from;
+        this.messageId = messageId;
+        this.phoneNumberId = phoneNumberId;
+        this.requestType = requestType;
+        this.status = status;
+        this.text = text;
+        this.to = to;
+    }
+}
+
 @RestController
 public class SmsHelloWorld {
 
   @RequestMapping(value = { "/incomingSms" }, method = RequestMethod.POST)
-  public void inboundSms(@RequestBody String body) {
-    String accountId = System.getenv("ACCOUNT_ID");
-    String apiKey = System.getenv("API_KEY");
-    String fromNumber = System.getenv("FREECLIMB_PHONE_NUMBER");
-    String toNumber = System.getenv("TO_PHONE_NUMBER");
+  public void inboundSms(@RequestBody IncomingSmsRequest request) {
+    Dotenv dotenv = Dotenv.configure().ignoreIfMissing().load();
+    String accountId = dotenv.get("ACCOUNT_ID");
+    String apiKey = dotenv.get("API_KEY");
+    String fromNumber = dotenv.get("FREECLIMB_NUMBER");
```

Do we want to throw an error if none of the environment variables are set, like how we are doing this check in the Node SMS quickstart and standardize this check across all language SMS quickstarts?

https://github.com/FreeClimbAPI/Node-SMS-Quickstart/pull/11/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R34